pkg/common/util/v1/testutil/pod.go

```diff
@@ -27,8 +27,8 @@ import (
 
 const (
 	// labels for pods and servers.
-	tfReplicaTypeLabel  = "tf-replica-type"
```

Comment:
For test, I think this is fine. We should be cautious for public facing labels. I think some users may have logics depends on it.

Reply:
Actually, since we now use `reconcilePod` from common library, the labels are defined in common library
https://github.com/kubeflow/common/blob/master/pkg/apis/common/v1/constants.go#L5
and added to the pods here
https://github.com/kubeflow/common/blob/master/pkg/controller.v1/common/pod.go#L436
As you can see, it is hard coded now. 
That means, no matter in tf-operator or mxnet-operator, the labels in the pod will always be `apiv1.ReplicaIndexLabel = "replica-index"` and not the one defined in each operator.
I think we should modify the common library to use the `GetReplicaTypeLabelKey`
```
func (tc *TFController) GetReplicaTypeLabelKey() string {
	return tfReplicaTypeLabel
}
```

Reply:
I created separate issue to track this https://github.com/kubeflow/common/issues/98

Reply:
I think we can have an interface there, and leave `replica-type` as default implementation in `kubeflow/common`, different operators can still override it if there's a need. 

This seems like a migration issue for tf only. Technically, we only use this label (and others) to filter pods with prefilter type. even they have same value, it's not a problem and won't affect filtering logic.

Reply:
Will it affect Katib? /cc @andreyvelich

Reply:
Yes, in Katib we use labels to search for master pod.

@Jeffwan @terrytangyuan In the new `common` implementation, it still applies `job-role: master` for master pod ?

Reply:
Yes for job-role label, it is still `job-role:master`
But for replica type label, it has changed and removed the tf-  prefix
I will solve the issue in common library so we can overwrite the replica type label 
 and then change this back in tf-operator

Reply:
Thanks @ChanYiLin! I believe it should work for us.